### PR TITLE
[saas-namespace-selector] SaaS File namespaceSelector

### PR DESCRIPTION
**Pull request by chassing**
Introduce dynamic selection of namespace (`resourceTemplates.targets.namespaceSelector`) in saas files.

Design doc: https://gitlab.cee.redhat.com/service/app-interface/-/merge_requests/59689

Ticket: [APPSRE-7124](https://issues.redhat.com/browse/APPSRE-7124)

### CODE REVIEW

File: graphql-schemas/schema.yml [hunk 1]
```diff
@@ -74,7 +74,7 @@ confs:
   - { name: schema, type: string, isRequired: true }
   - { name: path, type: string, isRequired: true }
   - { name: labels, type: json }
-  - { name: name, type: string, isRequired: true }
+  - { name: name, type: string, isRequired: true, isUnique: true }
```
Comment (maorfr) [inline graphql-schemas/schema.yml:77]: unrelated change?

**Comment by maorfr** [comment]
would you mind adding some schema level validation for "either namespace or namespace selector"?

File: graphql-schemas/schema.yml (hunk 1 above)
Comment (chassing) [reply graphql-schemas/schema.yml:77]: Yes, it's unrelated 🤦

**Comment by chassing** [comment]
> would you mind adding some schema level validation for "either namespace or namespace selector"?

How can that be done? I was looking for an example, but I need help finding one.

**Comment by maorfr** [comment]
you can start with `oneOf`?

**Comment by chassing** [comment]
> you can start with `oneOf`?

TIL `oneOf.required`. Thx

**Comment by chassing** [comment]
@maorfr a final LGTM would be appreciated 🙇

**Review by maorfr — APPROVED** [review]
